commonutils/src/main/java/io/intrepid/commonutils/StringUtils.java

```diff
@@ -0,0 +1,504 @@
+     *
+     * @param string      The string to repeat.
+     * @param separator   The separator between repetitions.
+     * @param repeatCount The times that the given string should be repeated.
+     * @return The repeated string.
+     */
+    public static CharSequence repeat(CharSequence string, CharSequence separator, int repeatCount) {
+        if (repeatCount < 1) {
+            return null;
+        }
+
+        if (repeatCount == 1) {
+            return string;
+        }
+
+        if (TextUtils.isEmpty(string) && TextUtils.isEmpty(separator)) {
+            return string;
+        }
+
+        if (string == null) {
+            string = "";
+        }
+
+        if (separator == null) {
+            separator = "";
+        }
+
+        final boolean stringIsNotEmpty = string.length() > 0;
+        final boolean separatorIsNotEmpty = separator.length() > 0;
+
+        final StringBuilder sb = new StringBuilder((string.length() + separator.length()) * repeatCount);
+
+        for (int i = 0; i < repeatCount; i++) {
+            if (separatorIsNotEmpty && (i > 0)) {
+                sb.append(separator);
+            }
+
+            if (stringIsNotEmpty) {
+                sb.append(string);
+            }
+        }
+
+        return sb.toString();
+    }
+
+    /**
+     * Calculates the Levenshtein Distance between two string, i.e. how many edits/changes would it
+     * take to get from string1 to string2.
+     * http://en.wikibooks.org/wiki/Algorithm_Implementation/Strings/Levenshtein_distance
+     *
+     * @param string1 The first string.
+     * @param string2 The second string.
+     * @return The distance between the two strings.
+     */
+    public static int getLevenshteinDistance(CharSequence string1, CharSequence string2) {
+        int len1 = string1.length() + 1;
+        int len2 = string2.length() + 1;
+
+        // the array of distances
+        int[] cost = new int[len1];
+        int[] newcost = new int[len1];
+
+        // initial cost of skipping prefix in String string1
+        for (int i = 0; i < len1; i++) cost[i] = i;
+
+        // dynamicaly computing the array of distances
+
+        // transformation cost for each letter in string2
+        for (int j = 1; j < len2; j++) {
+            // initial cost of skipping prefix in String string2
+            newcost[0] = j;
+
+            // transformation cost for each letter in string1
+            for (int i = 1; i < len1; i++) {
+                // matching current letters in both strings
+                int match = (string1.charAt(i - 1) == string2.charAt(j - 1)) ? 0 : 1;
+
+                // computing cost for each transformation
+                int cost_replace = cost[i - 1] + match;
+                int cost_insert = cost[i] + 1;
+                int cost_delete = newcost[i - 1] + 1;
+
+                // keep minimum cost
+                newcost[i] = Math.min(Math.min(cost_insert, cost_delete), cost_replace);
```

Comment:
Yes. We should replace this line with a `clamp` or some similar method when a MathUtils is added.